cypress/integration/providers_spec.js

```diff
@@ -0,0 +1,112 @@
+import { shouldBeRedirectedToLogin } from '../support/assertions';
+
+describe('Dashboard Providers Page', () => {
+  describe('When not logged in', () => {
+    it('should redirect to login page', () => {
+      cy.visit('/#/providers');
+      shouldBeRedirectedToLogin();
+    });
+  });
+
+  describe('When logged in', () => {
+    beforeEach(() => {
+      cy.login();
+      cy.task('resetState');
+    });
+
+    after(() => {
+      cy.task('resetState');
+    });
+
+    it('displays a link to view providers', () => {
+      cy.visit('/');
+
+      cy.contains('nav li a', 'Providers').should('exist').as('providers');
+      cy.get('@providers').should('have.attr', 'href', '#/providers');
+      cy.get('@providers').click();
+
+      cy.url().should('include', 'providers');
+      cy.get('.heading--xlarge').should('have.text', 'Providers');
+
+      cy.get('table tbody tr').its('length').should('be.eq', 2);
+    });
+
+    it('providers page displays a button to add a new provider', () => {
+      const name = 'TESTPROVIDER';
+      cy.visit('/#/providers');
+
+      cy.get('.heading--large').should('have.text', 'Provider Overview');
+      cy.contains('a', 'Add a Provider').should('exist').as('addProvider');
+      cy.get('@addProvider').should('have.attr', 'href', '#/providers/add');
+      cy.get('@addProvider').click();
+
+      cy.get('.heading--xlarge').should('have.text', 'Providers');
+      cy.get('.heading--large').should('have.text', 'Create a provider');
+
+      // fill the form and submit
+      cy.get('form div ul').as('providerinput');
+      cy.get('@providerinput').contains('Provider Name').siblings('input').type(name);
+      cy.get('@providerinput').contains('Concurrent Connnection Limit').siblings('input').clear().type(5);
+      cy.get('@providerinput').contains('label', 'Protocol').siblings().children('select')
+        .select('s3', {force: true}).should('have.value', 's3');
+
+      cy.get('@providerinput').contains('Host').siblings('input').type('test-host');
+
+      cy.get('form div input[value=Submit]').click();
+
+      // displays the new provider
+      cy.get('.heading--xlarge').should('have.text', 'Providers');
+      cy.get('.heading--large').should('have.text', name);
+      cy.url().should('include', `#/providers/provider/${name}`);
```

Comment:
Can we add some assertions for the specific provider property values?

Reply:
added assertion for property values of collection and provider